docs/generate_docs.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+
+import shutil
+import subprocess
+from pathlib import Path
+
+
+def run_command(command: list[str]) -> None:
+    """Run a command and exit if it fails."""
+    print(f"Running: {' '.join(command)}")
+    try:
+        subprocess.run(command, check=True)
+    except subprocess.CalledProcessError as e:
+        print(f"Warning: Command failed: {' '.join(command)}")
+        print(f"Error: {e}")
+
+
+def main():
+    docs_dir = Path(__file__).parent
+    project_root = docs_dir.parent
+
+    # Install documentation requirements
+    requirements_file = docs_dir / "requirements.txt"
+    run_command(["pip", "install", "-r", str(requirements_file)])
+
+    # Install core package
+    run_command(["pip", "install", "-e", "."])
+
+    # Install all service dependencies
```

Comment:
I opted to install all of the modules to avoid mocking every module. This made things vastly simpler.